### DISCUSSION TITLE
Place an object at a specific point

### DISCUSSION THREAD

**Discussion by Pandicon** [emilk/egui#1318] [Q&A]
I was wondering if it was possible to place a UI element at a specific point allowing it to overlay other elements. The docs and the [video tutorial](https://www.youtube.com/watch?v=NtUkr_z7l84) linked in the docs both create the UI by placing elements one after another and they can't overlap.
What I'm trying to do is to place elements at specific x and y coordinates even if the elements will overlap (for example if I want a circle to follow the mouse cursor, I would want it rendered at mouse x and y coordinates on top of other elements).
Is there a way to achieve this with egui?

**Comment by emilk** [comment]
If you just want to paint a circle, use `ui.painter()`. If you want to place a widget, use `ui.put` or `ui.allocate_ui_at_rect`.
